modules/platform/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/routing/WorkflowObject.java

```diff
@@ -67,12 +67,17 @@ protected void initialize(Object... args) {
 
     @POST
     public Response createWorkflowInstance(WorkflowRequest workflowRequest) {
-        final String workflowInstanceId = documentRoutingService.createNewInstance(
-                workflowRequest.getWorkflowModelName(), workflowRequest.getAttachedDocumentIds(),
-                workflowRequest.getVariables(), ctx.getCoreSession(), true);
-        DocumentModel workflowInstance = getContext().getCoreSession().getDocument(new IdRef(workflowInstanceId));
-        DocumentRoute route = workflowInstance.getAdapter(DocumentRoute.class);
-        return Response.ok(route).status(Status.CREATED).build();
+        if (documentRoutingService.canCreateInstance(ctx.getCoreSession(), workflowRequest.getAttachedDocumentIds(),
```

Comment:
Please put `ctx.getCoreSession()`, `workflowRequest.getAttachedDocumentIds()` and `workflowRequest.getWorkflowModelName()`, which are used several times, in local variables.
`getContext().getCoreSession()` could also use the same session local variable.

Reply:
done